packages/backend/src/models/ChangesetModel.ts

```diff
@@ -69,4 +84,57 @@ export class ChangesetModel {
         }
         return parsed.data;
     }
+
+    /**
+     * Creates a change in the database, if no active changeset exists, it will be created.
+     * @param params - The parameters for the change
+     * @returns The created change
+     */
+    async createChange(params: CreateChangeParams): Promise<DbChange> {
+        return this.database.transaction(async (trx) => {
+            let activeChangeset = await trx(ChangesetsTableName)
+                .where('project_uuid', params.projectUuid)
+                .orderBy('created_at', 'desc')
+                .first();
+
+            // Create changeset if it doesn't exist
+            if (!activeChangeset) {
+                const [newChangeset] = await trx(ChangesetsTableName)
+                    .insert({
+                        project_uuid: params.projectUuid,
+                        created_by_user_uuid: params.createdByUserUuid,
+                        updated_by_user_uuid: params.createdByUserUuid,
+                        status: 'draft',
+                        // TODO: Handle name in the future - auto generated for now.
+                        name: 'Auto-generated changeset',
+                    })
+                    .returning('*');
+
+                activeChangeset = newChangeset;
+            }
+
+            const changeParsed = DbChangeInsertSchema.safeParse({
+                changeset_uuid: activeChangeset.changeset_uuid,
+                created_by_user_uuid: params.createdByUserUuid,
+                source_prompt_uuid: params.sourcePromptUuid,
+                type: params.type,
+                entity_type: params.entityType,
+                entity_explore_uuid: params.entityExploreUuid,
+                entity_name: params.entityName,
+                payload: params.payload,
+            });
+
+            if (!changeParsed.success) {
+                throw new ParseError('Failed to parse change', {
+                    changeData: params,
+                });
+            }
+
+            const [change] = await trx(ChangesTableName)
+                .insert(changeParsed.data)
+                .returning('*');
+
+            return change;
```

Comment:
should this be returning parsed (transformed) entity? to metch the other model function?